KarhooUISDK/UIComponents/LegalNotice/KarhooLegalNoticeView.swift

```diff
@@ -0,0 +1,135 @@
+    
+    let shouldBeDisplayed: Bool = UITexts.Booking.legalNoticeText.isNotEmpty
+    private weak var parentViewController: UIViewController?
+    private let legalNoticeMailComposer: KarhooLegalNoticeMailComposerProtocol
+    private var zeroHeightTextConstreint: NSLayoutConstraint
+    
+    private var didSetUpConstraints: Bool = false
+    private var legalNoticeTextView: UITextView!
+    
+        private lazy var legalNoticeButton: KarhooExpandViewButton = {
+            let button = KarhooExpandViewButton(title: UITexts.Booking.legalNotice, onExpand: hideLegalNoticePressed, onCollapce: showLegalNoticePressed)
+            button.accessibilityIdentifier = KHLegalNoticeViewID.button
+            button.anchor(height: 44.0)
+            return button
+        }()
+    
+    private var attributedLabel: UILabel = {
+        let legalNoticeLabel = UILabel()
+        legalNoticeLabel.translatesAutoresizingMaskIntoConstraints = false
+        legalNoticeLabel.accessibilityIdentifier = KHLegalNoticeViewID.text
+        legalNoticeLabel.font = KarhooUI.fonts.captionRegular()
+        legalNoticeLabel.textAlignment = .justified
+        legalNoticeLabel.attributedText = LegalNoticeStringBuilder().legalNotice()
+        legalNoticeLabel.numberOfLines = 0
+        legalNoticeLabel.isUserInteractionEnabled = true
+        return legalNoticeLabel
+    }()
+    
+    // MARK: - Init
+    init(parent: UIViewController, linkParser: LinkParser = LinkParser()) {
+        zeroHeightTextConstreint = attributedLabel.heightAnchor.constraint(equalToConstant: 0)
+        self.parentViewController = parent
+        self.legalNoticeMailComposer = KarhooLegalNoticeMailComposer(parent: parentViewController)
+        super.init(frame: .zero)
+        self.setUpView()
+    }
+    
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup
+    private func setUpView() {
+        accessibilityIdentifier = KHLegalNoticeViewID.view
+        translatesAutoresizingMaskIntoConstraints = false
+        
+        attributedLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.singleTap(_:))))
+        
+        addSubview(legalNoticeButton)
+        addSubview(attributedLabel)
+        hideLegalNoticePressed()
+    }
+    
+    override func updateConstraints() {
+        if !didSetUpConstraints {
+            legalNoticeButton.anchor(
+                top: topAnchor,
+                leading: leadingAnchor,
+                bottom: attributedLabel.topAnchor,
+                paddingLeft: 8.0,
+                paddingRight: 8.0)
+            
+            attributedLabel.anchor(
+                top: legalNoticeButton.bottomAnchor,
+                leading: leadingAnchor,
+                bottom: bottomAnchor,
+                trailing: trailingAnchor,
+                paddingLeft: 8.0,
+                paddingBottom: 16,
+                paddingRight: 8.0)
```

Comment:
1. Closing bracket should be placed in new line
2. Are you sure you can not use any value from `UIConstants`? 8 and 16 look kinda standard. This applies to the whole PR.